ddtrace/_trace/tracing_plugins/__init__.py

```diff
@@ -0,0 +1,115 @@
+"""
+Integration tracing plugins.
+
+This module provides the plugin system for tracing integrations.
+
+The plugin system uses a hierarchical structure:
+
+    TracingPlugin (base)
+    ├── OutboundPlugin (TO external services)
+    │   ├── ClientPlugin (HTTP clients, etc.)
+    │   │   └── StoragePlugin (storage systems)
+    │   │       └── DatabasePlugin (databases)
+    │   └── ProducerPlugin (message producers)
+    └── InboundPlugin (FROM external sources)
+        ├── ServerPlugin (HTTP servers, etc.)
+        └── ConsumerPlugin (message consumers)
+
+Usage:
+    # During ddtrace initialization
+    from ddtrace._trace.tracing_plugins import initialize
```

Comment:
Are the plugins going to be tracer specific? Would it make sense to put these modules in `ddtrace/internal/plugins/....`. I can see other products (asm, llmobs, serverless, ...) extending or using these plugins. Making them tracing specific puts us in a corner.